homeassistant/components/tradfri/__init__.py

```diff
@@ -142,15 +142,16 @@ async def async_keep_alive(now: datetime) -> None:
 
         gw_status = True
         try:
-            await api(gateway.get_gateway_info())
-        except RequestError:
-            _LOGGER.error("Keep-alive failed")
+            await api(gateway.get_gateway_info(), timeout=TIMEOUT_API)
+        except PytradfriError:
```

Comment:
Where does the reload happen?

Reply:
in pytradfri (v7.20.0). This is by design, since the integration should have as little as possible protocol knowledge.

Reply:
Ok. Is there a reason we change the catch exception?

Reply:
its simpler, because it includes both request and timeout errors.

Reply:
The timeout error inherits RequestError.

https://github.com/home-assistant-libs/pytradfri/blob/e1d2d76cf189dfc4e154c8016c44cecaf1020ee5/pytradfri/error.py#L16

Reply:
Yes but I am moving the whole integration in direction of using pytradfriError consistently.

Reply:
Why? Normally it's best to catch as specific error as possible. If we want to add another error in pytradfri and have the integration react differently to that error we would need to change this again.

Reply:
ok.

Reply:
`RequestError` is appropriate since all errors raised by the api call that are defined by pytradfri inherit from `RequestError`. It's exactly the purpose of this error to be used when handling api call exceptions.